pkg/controller/template/template_controller.go

```diff
@@ -103,6 +112,65 @@ func New(
 	return ctrl
 }
 
+func (ctrl *Controller) addSecret(obj interface{}) {
+	s := obj.(*v1.Secret)
+	if s.DeletionTimestamp != nil {
+		ctrl.deleteSecret(s)
+		return
+	}
+	if s.GetNamespace() == "openshift-config" && s.GetName() == "pull-secret" {
+		cfg, err := ctrl.ccLister.Get(common.ControllerConfigName)
+		if err != nil {
+			utilruntime.HandleError(fmt.Errorf("Couldn't get ControllerConfig on secret callback %#v", err))
+			return
+		}
+		glog.V(4).Infof("Re-syncing ControllerConfig %s due to secret change", cfg.Name)
+		ctrl.enqueueControllerConfig(cfg)
+	}
+}
+
+func (ctrl *Controller) updateSecret(old, new interface{}) {
+	s := new.(*v1.Secret)
+
+	if s.GetNamespace() == "openshift-config" && s.GetName() == "pull-secret" {
+		cfg, err := ctrl.ccLister.Get(common.ControllerConfigName)
+		if err != nil {
+			utilruntime.HandleError(fmt.Errorf("Couldn't get ControllerConfig on secret callback %#v", err))
+			return
+		}
+		glog.V(4).Infof("Re-syncing ControllerConfig %s due to secret change", cfg.Name)
+		ctrl.enqueueControllerConfig(cfg)
+	}
+}
+
+func (ctrl *Controller) deleteSecret(obj interface{}) {
+	s, ok := obj.(*v1.Secret)
+
+	if !ok {
+		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			utilruntime.HandleError(fmt.Errorf("Couldn't get object from tombstone %#v", obj))
+			return
+		}
+		s, ok = tombstone.Obj.(*v1.Secret)
+		if !ok {
+			utilruntime.HandleError(fmt.Errorf("Tombstone contained object that is not a MachineConfig %#v", obj))
+			return
+		}
+	}
+
+	if s.GetNamespace() == "openshift-config" && s.GetName() == "pull-secret" {
```

Comment:
since all 3 funcs seem to use the same if-if-block  maybe just move it out to have 1 as a func and call that?

Reply:
done, though, for the delete action this is not the case as that's not enqueuing anything but I've added a TODO for later.

Reply:
nice!